hwmv2: port intel x86 boards

golowanow opened the pull request:
move and convert x86 soc/boards to HWMv2:

* atom

* apollo_lake
  * up_squared

* alder_lake
  * intel_adl_crb
  * intel_adl_rvp
  *  up_squared_pro_7000

* elkhart_lake
  * intel_ehl_crb
  * intel_ehl_crb//sbl

* raptor_lake
  * intel_rpl_p_crb
  * intel_rpl_s_crb

golowanow committed:
boards: x86: up_squared: move and convert to HWMv2

Move and convert boards/x86/up_squared to HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
samples: up_squared: adjust gpio_counter to HWMv2

Adjust samples/boards/up_squared/gpio_counter to HWMv2 changes
of the up_squared board.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: up_squared_pro_7000: move and convert to HWMv2

Move and convert boards/x86/up_squared_pro_7000 to HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: intel_adl: move and convert alder_lake boards to HWMv2

Move and convert intel_adl_crb and intel_adl_rvp boards to HWMv2.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nashif commented on soc/v2/intel/intel_x86/soc.yml:
```diff
@@ -0,0 +1,2 @@
+family:
+- name: intel_x86
```
This might be a bit too generic and everything would fall under it, i.e. it almost matches the architecture and not soc families. There should be a way to define SoCs without families if I am not mistaken, ISH for example could be declared as family. We also have legacy stuff (ia32) that would need to be dealt up with and potentially cleaned up/dropped.

golowanow replied:
changed to have no 'family' or 'series' for these SoC

golowanow committed:
soc: apollo_lake: move and convert to HWMv2

Move and convert soc/x86/apollo_lake to HWMv2.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: alder_lake: move and convert to HWMv2

Move and convert soc/x86/alder_lake to HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: x86: elkhart_lake: move and convert to HWMv2

Move and convert soc/x86/elkhart_lake ot HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: intel_ehl: move and convert elkhart_lake boards to HWMv2

Move and convert `intel_ehl_crb` and `intel_ehl_crb_sbl` board
configurations to HWMv2.

`intel_ehl_crb_sbl` is changed to a variant `intel_ehl_crb/elkhart_lake/sbl`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: x86: raptor_lake: move and convert to HWMv2

Move and convert soc/x86/raptor_lake to HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: intel_rpl: move and convert raptor_lake boards to HWMv2

Move and convert intel_rpl_p_crb and intel_rpl_s_crb boards to HWMv2.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: x86: atom: move and convert to HWMv2

Move and convert soc/x86/atom to HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented on soc/v2/ia32/soc.yml:
`ia32` is put at the top level as generic, please confirm is it ok ?

golowanow replied:
moved to #68078

finikorg commented:
Should the vendor name be `up` or `aaeon` ?

golowanow commented:
> Should the vendor name be `up` or `aaeon` ?

good question, it is disributed by ['up bridge the gap'](https://up-board.org/upsquared/specifications/) and at #67682 it is listed as 'up'.

golowanow commented:
SoC: ia32 and lakemont are moved to #68078 to complement qemu_x86

golowanow commented:
CI twister run has no issues after the rebase

golowanow commented:
@tejlmand - could you please provide some feedback - is this PR on track with the general v1 to v2 migration principles ?

golowanow commented on boards/v2/intel/intel_ehl/board.yml:
```diff
@@ -0,0 +1,9 @@
+    vendor: intel
+    socs:
+      - name: elkhart_lake
+  - name: intel_ehl_crb_sbl
+    vendor: intel
+    socs:
+      - name: elkhart_lake
```
@nashif  should we also optimize boards to variants doing migration or keep them as-as for now ?

nashif replied:
yes, they should be converted to variants now.

golowanow replied:
`intel_ehl_crb_sbl` converted to `intel_ehl_crb//sbl` variant

golowanow commented on boards/v2/intel/intel_rpl/board.yml:
```diff
@@ -0,0 +1,9 @@
+boards:
+  - name: intel_rpl_p_crb
+    vendor: intel
+    socs:
+      - name: raptor_lake
+  - name: intel_rpl_s_crb
+    vendor: intel
+    socs:
+      - name: raptor_lake
```
@nashif should we migrate these boards to variants or keep them as-is ?

golowanow replied:
keeping these as different boards
https://docs.zephyrproject.org/latest/boards/x86/intel_rpl/doc/index.html

tejlmand commented on soc/v2/intel/elkhart_lake/Kconfig.soc:
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_ELKHART_LAKE
+	bool
```
you should move those two lines from the defconfig to this file:
https://github.com/zephyrproject-rtos/zephyr/blob/39ab37dd7bae8d56a6e2fd4a3d1493e433c01559/soc/v2/intel/elkhart_lake/Kconfig.defconfig#L9-L10

golowanow replied:
done

tejlmand reviewed (changes requested):
Looks good, only minor adjustment needed.

Also consider if some of the boards should be using variants as pointed out by @nordicjm

golowanow commented on boards/v2/intel/intel_adl/Kconfig.defconfig:
```diff
@@ -58,4 +51,7 @@ endif
+config HAS_COVERAGE_SUPPORT
+	default y
+
+#
```
removed

golowanow commented on boards/v2/up/up_squared_pro_7000/Kconfig.defconfig:
```diff
@@ -0,0 +1,50 @@
+if BOARD_UP_SQUARED_PRO_7000
+
+config BOARD
+	default "up_squared_pro_7000"
```
removed

golowanow commented on boards/v2/intel/intel_rpl/Kconfig.defconfig:
```diff
@@ -62,4 +57,7 @@ config SHELL_STACK_SIZE
+config HAS_COVERAGE_SUPPORT
+	default y
+
+#
```
done
